src/routes/things.js

```diff
@@ -47,8 +57,104 @@ let router = express.Router();
  * }
  **/
 
-router.get('/Things', (req, res) => {
-  res.status(200).send();
+const resource = 'Things';
+
+router.get('/', (req, res) => {
+  db().then((models) => {
+    if (req.params && req.params[0]) {
+      models.Things.findById(req.params[0], {
+        attributes: {
+          exclude: excludedFields
+        }
+      }).then(thing => {
+        res.status(200).json(response.generate(thing));
+      }).catch(() => {
+        ApiError(res, 404, ERRNO_RESOURCE_NOT_FOUND, BAD_REQUEST);
+      });
+    } else {
+      models.Things.findAll({
+        attributes: {
+          exclude: excludedFields
+        }
+      }).then(things => {
+        res.status(200).json(response.generate(things));
+      });
+    }
+  }).catch(() => {
+    ApiError(res, 500, ERRNO_INTERNAL_ERROR, INTERNAL_ERROR);
+  });
+});
+
+// XXX #9 [Things router] Implement associations to other models and integrity
+
+router.post('/', (req, res) => {
+  db().then((models) => {
+    models.Things.create(req.body)
+    .then(createdThing => {
+
+      // XXX #13 [Things router] response urls should be absolute
+
+      res.location('/' + resource + '/' + createdThing.id);
+      res.status(201).send();
+    }).catch(() => {
+      res.status(400).send();
+    });
+  }).catch(() => {
+    ApiError(res, 500, ERRNO_INTERNAL_ERROR, INTERNAL_ERROR);
+  });
+});
+
+// XXX #10 [Things router] Ignore entity id if present in PATCH request
+// XXX #11 [Things router] Check that entity does not contain related entities
+// as inline content in PATCH requests
+// XXX #12 [Things router] Handle navigation properties in PATCH requests
+
+router.patch('/', (req, res) => {
+  if (!req.params || !req.params[0]) {
+
+    return ApiError(res, 404, ERRNO_RESOURCE_NOT_FOUND, BAD_REQUEST);
+  }
+
+  db().then((models) => {
+    models.Things.findById(req.params[0], {
```

Comment:
Do we need to search before update? You can probably do something like:

`models.Things.update(req.body, { where: { id: req.params[0] } })`

Make sure you handle the case where id is invalid (404). Not sure if that's an exception (.catch) or if it causes the resolution of the promise with an empty array or null.